mne/report/report.py

```diff
@@ -1191,27 +1529,115 @@ class construction.
+        warn(
+            message='`Report.add_figs_to_section` has been deprecated. Use '
+                    '`Report.add_figure` instead',
+            category=DeprecationWarning
+        )
+        figs, captions, comments = self._validate_input(
+            figs, captions, section, comments
+        )
         image_format = _check_image_format(self, image_format)
         _check_scale(scale)
-        for fig, caption, comment in zip(figs, captions, comments):
-            caption = 'custom plot' if caption == '' else caption
-            sectionvar = self._sectionvars[section]
-            global_id = self._get_id()
-            div_klass = self._sectionvars[section]
-            img_klass = self._sectionvars[section]
-
-            img = _fig_to_img(fig, image_format, scale, auto_close)
-            html = image_template.substitute(img=img, id=global_id,
-                                             div_klass=div_klass,
-                                             img_klass=img_klass,
-                                             caption=caption,
-                                             show=True,
-                                             image_format=image_format,
-                                             comment=comment)
-            self._add_or_replace('%s-#-%s-#-custom' % (caption, sectionvar),
-                                 sectionvar, html, replace)
+
+        if (
+            _check_path_like(figs) or
+            (hasattr(figs, '__iter__') and
+             any(_check_path_like(f) for f in figs))
+        ):
+            raise TypeError(
+                'It seems you passed a path to `add_figs_to_section`. '
+                'However, only Matplotlib figures, Mayavi scenes, and NumPy '
+                'arrays are accepted. You may want to try `add_image` instead.'
+            )
+
+        if hasattr(figs, '__len__') and not isinstance(figs, np.ndarray):
+            figs = tuple(figs)
+        else:
+            figs = (figs,)
+
+        if isinstance(captions, str):
+            captions = (captions,)
+        else:
+            captions = tuple(captions)
+
+        if isinstance(comments, str):
+            comments = (comments,)
+        elif comments is None:
+            comments = (None,) * len(figs)
+        else:
+            comments = tuple(comments)
+
+        if len(figs) != len(captions):
+            raise ValueError(
+                f'Number of figs ({len(figs)}) must equal number of captions '
+                f'({len(captions)})'
+            )
+        if len(figs) != len(comments):
+            raise ValueError(
+                f'Number of figs ({len(figs)}) must equal number of comments '
+                f'({len(comments)})'
+            )
+
+        tags = _clean_tags(section)
+
+        for fig, title, caption in zip(figs, captions, comments):
+            if scale is not None:
+                _scale_mpl_figure(fig, scale)
+            tight_layout(fig=fig)
```

Comment:
So far I've observed loss of functionality in `add_figs_to_section`, it used to support `figs` as a `np.ndarray` directly (e.g., a `(h, w, 3)` array) directly and now I get:
```
  File "/home/larsoner/mne_data/mnefun/mnefun/_report.py", line 594, in gen_html_report
    report.add_figs_to_section(view, captions, section)
  File "/home/larsoner/python/mne-python/mne/report/report.py", line 1586, in add_figs_to_section
    image_format=image_format, tags=tags, replace=replace
  File "/home/larsoner/python/mne-python/mne/viz/utils.py", line 141, in tight_layout
    fig.canvas.draw()
AttributeError: 'numpy.ndarray' object has no attribute 'canvas'
```
(line numbers changed above because I'm making other changes)

I could push a fix that checks if it's an instance of `matplotlib.figure.Figure`, but I also don't like that this `tight_layout` is being forced. I think users should do this at their own end if they want it.

Moreover, I get warnings like this because it's being forced on me:
```
  report.add_slider_to_section(
/home/larsoner/mne_data/mnefun/mnefun/_report.py:943: UserWarning: tight_layout not applied: number of columns in subplot specifications must be multiples of one another.
```
So I'm going to remove this...

Reply:
Passing an array SHOULD work, if it doesn't then it's a bug
I can have a look at this

Reply:
Don't worry, I'm pushing a commit to remove `tight_layout` entirely here